plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/StandardFacetInstallDelegate.java

```diff
@@ -34,18 +34,32 @@
 import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
 
 public class StandardFacetInstallDelegate extends AppEngineFacetInstallDelegate {
-  private final static String APPENGINE_WEB_XML = "appengine-web.xml";
+  private static final String APPENGINE_WEB_XML = "appengine-web.xml";
 
   @Override
   public void execute(IProject project,
                       IProjectFacetVersion version,
                       Object config,
                       IProgressMonitor monitor) throws CoreException {
     super.execute(project, version, config, monitor);
+    // If Dynamic Web facet was additionally installed together, it will have created "WebContent/"
+    // as the default webapp directory (unless the user changed the default when installing
+    // the Web facet. We detect if there already existed another webapp directory, and if so,
+    // fix the default webapp setting to point to the existing directory.
+    fixWebContentRootIfNecessary(project);
+
     createConfigFiles(project, monitor);
     installAppEngineRuntimes(project);
   }
 
+  private void fixWebContentRootIfNecessary(IProject project) {
+    if (WebProjectUtil.isDefaultWebContentFolderBogus(project)) {
+      Fix/update webapp setting in ".settings/org.eclipse.wst.common.component" (XML). How? To figure out...
```

Comment:
I don't think we can. For the new project wizard, it is us that freshly installs the Web facet programmatically with the code above. When a user is adding a facet (both the App Engine and Web facets) to an existing project, it is WTP that's doing the that. WTP sets "WebContent" by default.